SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -832,16 +832,13 @@ bool USpatialSender::WillHaveAuthorityOverActor(AActor* TargetActor, Worker_Enti
 {
 	bool WillHaveAuthorityOverActor = true;
 
-	if (GetDefault<USpatialGDKSettings>()->bEnableMultiWorker)
+	if (NetDriver->VirtualWorkerTranslator != nullptr)
 	{
-		if (NetDriver->VirtualWorkerTranslator != nullptr)
+		if (const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(TargetEntity))
 		{
-			if (const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(TargetEntity))
+			if (AuthorityIntentComponent->VirtualWorkerId != NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId())
 			{
-				if (AuthorityIntentComponent->VirtualWorkerId != NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId())
-				{
-					WillHaveAuthorityOverActor = false;
-				}
+				WillHaveAuthorityOverActor = false;
 			}
 		}
 	}
```

Comment:
I feel like this whole function might be redundant as of recently. Can we check authority role where this function is called instead> Checking intent this way seems like we might be overcomplicating it, authority role should encapsulate local changes to auth intent, and if it doesn't we might've done something wrong?

Reply:
So I put a break point in the false return and ran both our RPC unit tests and the only RPC gym we have, and neither hit the breakpoint. I spent a bit of time looking at the code, and I think you're right (and even if not, we just wouldn't drop a RPC, so the worst case is a performance drop) but I'm uncomfortable making this change right before 0.10. How would you feel about a follow-up ticket?